cli/lakeflags/datadir.go

```diff
@@ -0,0 +1,51 @@
+package lakeflags
+
+import (
+	"os"
+	"os/user"
+	"path/filepath"
+	"runtime"
+)
+
+var defaultDataDir string
+
+func init() {
+	defaultDataDir = getDefaultDataDir()
+}
+
+// getDefaultDataDir returns the default data directory for the current user.
+// Derived from https://github.com/btcsuite/btcd/blob/master/btcutil/appdata.go
+func getDefaultDataDir() string {
+	// Resolve the XDG data home directory if set.
+	if xdgDataHome := os.Getenv("XDG_DATA_HOME"); xdgDataHome != "" {
+		return filepath.Join(xdgDataHome, "zed")
+	}
+	// Get the OS specific home directory via the Go standard lib.
+	var homeDir string
+	usr, err := user.Current()
+	if err == nil {
+		homeDir = usr.HomeDir
+	}
+	// Fall back to standard HOME environment variable that works
+	// for most POSIX OSes if the directory from the Go standard
+	// lib failed.
+	if err != nil || homeDir == "" {
+		homeDir = os.Getenv("HOME")
+	}
+	switch runtime.GOOS {
+	case "windows":
+		if appData := os.Getenv("LOCALAPPDATA"); appData != "" {
+			return filepath.Join(appData, "zed")
+		}
+	default:
+		if homeDir != "" {
+			// For linux and windows follow the XDG spec which states:
+			// If $XDG_DATA_HOME is either not set or empty, a default equal to
+			// $HOME/.local/share should be used.
+			return filepath.Join(homeDir, ".local", "share", "zed")
+		}
+	}
```

Comment:
Maybe simplify this?
```suggestion
	if runtime.GOOS == "windows" {
		if appData := os.Getenv("LOCALAPPDATA"); appData != "" {
			return filepath.Join(appData, "zed")
		}
	}
	if homeDir, _ := os.UserHomeDir(); homeDir != "" {
		// Follow the XDG spec which states:
		// If $XDG_DATA_HOME is either not set or empty, a default equal to
		// $HOME/.local/share should be used.
		return filepath.Join(homeDir, ".local", "share", "zed")
	}
```